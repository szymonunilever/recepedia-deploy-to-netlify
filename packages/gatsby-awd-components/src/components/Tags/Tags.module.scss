@import '../../scss/config';

.tags {
  &__list {
    list-style: none;
    display: flex;
    flex-wrap: wrap;
    margin: -$tags-indent;
    padding: $tags-list-padding;
  }
  &__load-more {
    margin: $tags-indent;
  }
}

:global(.button.tags__button) {
  background: $tag-background;
  height: $tag-height;
  color: $tag-color;
  font-size: $tag-fz;
  display: inline-flex;
  align-items: center;
  justify-content: space-between;
  position: relative;
  padding: $tag-padding;
  font-weight: normal;
  transition: $theme-transition;
  cursor: pointer;
  text-decoration: none;
  border-radius: $tag-border-radius;
  border: none;
  @media (hover: hover),
    (-ms-high-contrast: none),
    (-ms-high-contrast: active) {
    &:hover {
      background: $tag-background-hover;
      color: $tag-color-hover;
    }
  }
}
