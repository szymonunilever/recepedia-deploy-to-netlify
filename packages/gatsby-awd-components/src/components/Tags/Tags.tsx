import cx from 'classnames';
import React, { useEffect, useState } from 'react';
import { Button } from '../Button';
import { TagsProps, TagViewType, TagVariant } from './models';
import Tag from './partials/Tag';
import includes from 'lodash/includes';
import uniqWith from 'lodash/uniqWith';
import { TagName, Text } from '../Text';
import isEqual from 'lodash/isEqual';
import getComponentDataAttrs from '../../utils/getComponentDataAttrs';
import theme from './Tags.module.scss';

const getTagList = (list: Internal.Tag[], displayUniq = true) =>
  displayUniq ?
    uniqWith(list, (item1, item2) => item1.fields.slug === item2.fields.slug) :
    list;

const Tags = ({
  list,
  content,
  tagsPerLoad = 4,
  initialCount = 4,
  RemoveIcon,
  handleTagToggle,
  enableExternalManage = false,
  handleTagRemove,
  selectedTags,
  className,
  viewType = TagViewType.standard,
  variant = TagVariant.link,
  titleLevel = 2,
  displayOnlyUniqueNames = true,
}: TagsProps) => {
  const { title, loadMoreButton } = content || {
    title: undefined,
    loadMoreButton: undefined,
  };

  const tagsList = getTagList(list, displayOnlyUniqueNames);

  const [tags, setTags] = useState({
    list: tagsList,
    displayList:
      initialCount !== 'all' ? tagsList.slice(0, initialCount) : tagsList,
  });

  useEffect(() => {
    const tagsListUpdated = getTagList(list, displayOnlyUniqueNames);

    setTags({
      list: tagsListUpdated,
      displayList:
        initialCount !== 'all'
          ? tagsListUpdated.slice(
              0,
              Math.max(tags.displayList.length, initialCount)
            )
          : tagsListUpdated,
    });
  }, [list, initialCount]);

  const loadMore = () => {
    const newCount = tags.displayList.length + tagsPerLoad;

    setTags({
      ...tags,
      displayList: tags.list.slice(0, newCount),
    });
  };

  const deleteItem = (tag: Internal.Tag) => {
    setTags({
      list: tags.list.filter((item: Internal.Tag) => item.id !== tag.id),
      displayList: tags.displayList.filter(
        (item: Internal.Tag) => item.id !== tag.id
      ),
    });
    if (handleTagRemove) {
      handleTagRemove(tag);
    }
  };

  useEffect(() => {
    if (enableExternalManage && !isEqual(list, tags.list)) {
      const tagsList = getTagList(list, displayOnlyUniqueNames);

      setTags({
        ...tags,
        list: tagsList,
        displayList:
          initialCount !== 'all' ? tagsList.slice(0, initialCount) : tagsList,
      });
    }
  }, []);

  const shouldAppear =
    tags.list.length > tags.displayList.length && loadMoreButton;

  const loadMoreBtn = shouldAppear ? (
    <li className={cx(theme.tags__loadMore, 'tags__load-more')}>
      <Button
        onClick={loadMore}
        className={cx(theme.tags__button, 'tags__button')}
      >
        <span>{loadMoreButton ? loadMoreButton.label : null}</span>
      </Button>
    </li>
  ) : null;

  const classNames = cx(theme.tags, 'tags', className);

  const view =
    viewType === TagViewType.filter ? (
      <div className={classNames} {...getComponentDataAttrs('tags', content)}>
        <ul className={cx(theme.tags__list, 'tags__list')}>
          {tags.list.map((item: Internal.Tag) => (
            <Tag
              key={item.id}
              handleClick={deleteItem}
              tag={item}
              enableExternalManage={enableExternalManage}
              active={includes(selectedTags, item)}
              variant={variant}
              handleToggle={handleTagToggle}
            />
          ))}
        </ul>
      </div>
    ) : (
      <div className={classNames} {...getComponentDataAttrs('tags', content)}>
        {title && (
          <Text
            className={cx(theme.tags__title, 'tags__title')}
            // @ts-ignore
            tag={TagName[`h${titleLevel}`]}
            text={title}
          />
        )}

        {!!tags.displayList.length && (
          <ul className={cx(theme.tags__list, 'tags__list')}>
            {tags.displayList.map((item: Internal.Tag) => (
              <Tag
                RemoveIcon={RemoveIcon}
                key={item.id}
                handleClick={deleteItem}
                tag={item}
                variant={variant}
              />
            ))}
            {loadMoreBtn}
          </ul>
        )}
      </div>
    );

  return <>{view}</>;
};

export default Tags;
