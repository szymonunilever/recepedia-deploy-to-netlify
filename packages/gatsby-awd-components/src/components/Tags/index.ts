export { default as Tags } from './Tags';
export { default } from './Tags';
export * from './models';
