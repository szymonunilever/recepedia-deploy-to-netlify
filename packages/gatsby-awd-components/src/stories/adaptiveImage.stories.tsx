import { storiesOf } from '@storybook/react';
import React from 'react';
import { AdaptiveImage } from '../index';
// import localImage from './assets/localImage';
import localImage from './assets/localImageMulti';
// import { findLastIndex } from 'lodash-es';

const image = {
  url:
    'https://scm-assets.constant.co/scm/unilever/1d398653b55393fd6da9bff8ea193338/bab4fc1b-c269-44c1-8d60-367626f8b029.jpg',
  alt: 'Alt',
};

storiesOf('Generic/AdaptiveImage', module)
  .add(
    'default',
    () => (
      <AdaptiveImage
        className="custom-class"
        localImage={localImage}
        alt={image.alt}
      />
    ),
    { inline: false }
  )
  .add('with link', () => (
    <AdaptiveImage
      className="custom-class"
      localImage={localImage}
      alt={image.alt}
      link={'/'}
    />
  ))
  .add('with sizes specified', () => (
    <div>
      <AdaptiveImage
        className="custom-class"
        localImage={localImage}
        alt={image.alt}
        sizes="(max-width: 320px) 100vw, 200px"
      />
      <p>
        Note: image width is set to 200px. If browser cache has the same image
        of bigger size, it will be used instead of small image.
      </p>
    </div>
  ));
