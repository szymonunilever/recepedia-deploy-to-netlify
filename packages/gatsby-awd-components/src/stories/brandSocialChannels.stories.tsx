import { storiesOf } from '@storybook/react';
import React from 'react';
import { BrandSocialChannels } from '../index';
import content from '../mocks/brandSocialChannels.json';
import { ReactComponent as FbIcon } from 'src/svgs/inline/facebook.svg';
import { ReactComponent as TwitterIcon } from 'src/svgs/inline/twitter.svg';
import { ReactComponent as InstaIcon } from 'src/svgs/inline/instagram.svg';

const list = {
  facebook: <FbIcon />,
  twitter: <TwitterIcon />,
  instagram: <InstaIcon />,
};

const config = {
  displayLabel: false,
};

storiesOf('Generic/BrandSocialChannels', module).add('default', () => (
  <BrandSocialChannels listIcons={list} content={content} {...config} />
));
