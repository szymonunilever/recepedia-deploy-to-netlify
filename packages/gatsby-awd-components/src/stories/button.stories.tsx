import { storiesOf } from '@storybook/react';
import React from 'react';
import { Button, ButtonViewType } from '../index';
import { ReactComponent as Facebook } from '../svgs/inline/facebook.svg';
import { ReactComponent as Facebook2 } from '../svgs/inline/facebook2.svg';

const onClick = () => alert('on click button action');

storiesOf('Generic/Button', module)
  .add('with text label', () => (
    <>
      <Button {...{ onClick }} className="button--small">
        small
      </Button>
      <Button {...{ onClick }} className="button--medium">
        medium
      </Button>
      <Button {...{ onClick }} className="button--large">
        large
      </Button>
    </>
  ))
  .add('with text label disabled', () => (
    <Button {...{ onClick }} className="custom-class" isDisabled>
      Click me
    </Button>
  ))
  .add('with icon viewType', () => (
    <div style={{ fontSize: '3rem' }}>
      <Button
        {...{ onClick }}
        Icon={Facebook}
        className="custom-class"
        viewType={ButtonViewType.icon}
      />
    </div>
  ))
  .add('with different icons for toggled states', () => (
    <div style={{ fontSize: '3rem' }}>
      <Button
        Icon={Facebook}
        IconSelected={Facebook2}
        className="custom-class"
        isToggle
        viewType={ButtonViewType.icon}
      />
    </div>
  ))
  .add('with icon and text', () => (
    <Button Icon={Facebook} {...{ onClick }} className="custom-class">
      Click me
    </Button>
  ));
