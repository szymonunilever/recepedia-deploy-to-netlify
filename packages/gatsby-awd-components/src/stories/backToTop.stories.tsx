import { storiesOf } from '@storybook/react';
import React from 'react';
import { ReactComponent as Icon } from 'src/svgs/inline/arrow-up.svg';
import { BackToTop } from '../index';
import { ButtonViewType } from '../components/Button';

const page = {
  height: '200vh',
  fontSize: '2rem',
};

const content: AppContent.BackToTopContent = {
  label: 'Back to top',
};

storiesOf('Generic/Back To Top', module)
  .add(
    'with Icon',
    () => (
      <>
        <div style={page}>
          Scroll down to see Back-to-top button in the right bottom corner
          <BackToTop content={{ label: undefined }} Icon={Icon} />
        </div>
      </>
    ),
    { inline: false }
  )
  .add(
    'with Icon and Text',
    () => (
      <>
        <div style={page}>
          Scroll down to see Back-to-top button in the right bottom corner
          <BackToTop content={content} Icon={Icon} />
        </div>
      </>
    ),
    { inline: false }
  )
  .add(
    'with Icon viewType',
    () => (
      <>
        <div style={page}>
          Scroll down to see Back-to-top button in the right bottom corner
          <BackToTop
            content={{ label: undefined }}
            Icon={Icon}
            viewType={ButtonViewType.icon}
          />
        </div>
      </>
    ),
    { inline: false }
  );
