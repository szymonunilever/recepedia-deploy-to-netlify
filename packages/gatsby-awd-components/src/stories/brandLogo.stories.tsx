import { storiesOf } from '@storybook/react';
import React from 'react';
import { BrandLogo } from '../components/BrandLogo';

const brandList = ['knorr', 'hellmanns', 'maizena'];

storiesOf('Generic/BrandLogo', module).add('Default view', () => (
  <>
    {brandList.map(brand => (
      <BrandLogo brandTheme={brand} linkTo={'/'} key={brand} />
    ))}
  </>
));
