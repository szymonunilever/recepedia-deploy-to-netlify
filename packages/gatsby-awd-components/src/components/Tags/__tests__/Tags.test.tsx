import { mount, ReactWrapper } from 'enzyme';
import React from 'react';
import Tags, { TagsProps } from '../index';
import toJson from 'enzyme-to-json';
import tagsData from '../../../mocks/tags.json';
import { config } from '../../../stories/tags.stories';

describe('<Tags />', () => {
  let wrapper: ReactWrapper;
  // @ts-ignore
  const tagsProps: TagsProps = {
    list: tagsData,
  };

  beforeEach(() => {
    wrapper = mount(<Tags {...tagsProps} {...config} />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('matches the snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
