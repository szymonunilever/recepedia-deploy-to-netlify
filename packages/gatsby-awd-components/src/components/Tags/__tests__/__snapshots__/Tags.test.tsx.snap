// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tags /> matches the snapshot 1`] = `
<Tags
  className="custom-class"
  content={
    Object {
      "loadMoreButton": Object {
        "label": "Custom button text",
      },
      "title": "Custom title text",
    }
  }
  initialCount={2}
  isEditable={true}
  list={
    Array [
      Object {
        "fields": Object {
          "slug": "/hub-de-conteudo/egg-free",
        },
        "id": "d717752e-4a33-54a7-9273-05763e8773ce",
        "name": "egg_free",
        "tagId": 2,
        "title": "Sem ovos",
      },
      Object {
        "fields": Object {
          "slug": "/hub-de-conteudo/gluten-free",
        },
        "id": "6661e4bb-4176-5612-8341-04e2a9379d09",
        "name": "gluten_free",
        "tagId": 3,
        "title": "Glúten Free",
      },
      Object {
        "fields": Object {
          "slug": "/hub-de-conteudo/lactose-free",
        },
        "id": "8e2456ac-d41a-542f-885d-c19ebe96d38a",
        "name": "lactose_free",
        "tagId": 4,
        "title": "Zero Lactose",
      },
      Object {
        "fields": Object {
          "slug": "/hub-de-conteudo/nut-free",
        },
        "id": "1f1a127f-8c54-56a2-903d-0e9d6096ad2d",
        "name": "nut_free",
        "tagId": 5,
        "title": "Livre de nozes",
      },
      Object {
        "fields": Object {
          "slug": "/hub-de-conteudo/paleo-diet",
        },
        "id": "79692d6c-840c-5a55-8907-8a5b1a605a67",
        "name": "paleo_diet",
        "tagId": 6,
        "title": "Paleo diet",
      },
      Object {
        "fields": Object {
          "slug": "/hub-de-conteudo/pregnancy-safe",
        },
        "id": "c7aa9d60-ef6d-5870-885a-bbdf6977072d",
        "name": "pregnancy_safe",
        "tagId": 7,
        "title": "Seguro de gravidez",
      },
    ]
  }
  tagsPerLoad={1}
>
  <div
    className="tags tags custom-class"
    data-componentname="tags"
    data-componentview="tags"
  >
    <Text
      className="tags__title tags__title"
      tag={1}
      text="Custom title text"
    >
      <h2
        className="tags__title tags__title"
      >
        Custom title text
      </h2>
    </Text>
    <ul
      className="tags__list tags__list"
    >
      <Tag
        handleClick={[Function]}
        key="d717752e-4a33-54a7-9273-05763e8773ce"
        tag={
          Object {
            "fields": Object {
              "slug": "/hub-de-conteudo/egg-free",
            },
            "id": "d717752e-4a33-54a7-9273-05763e8773ce",
            "name": "egg_free",
            "tagId": 2,
            "title": "Sem ovos",
          }
        }
        variant={1}
      >
        <li
          className="tags__item tags__item"
        >
          <mockConstructor
            className="tags__link tags__link"
            to="/hub-de-conteudo/egg-free"
          >
            <a
              className="tags__link tags__link"
              href="/hub-de-conteudo/egg-free"
            >
              <span>
                Sem ovos
              </span>
            </a>
          </mockConstructor>
        </li>
      </Tag>
      <Tag
        handleClick={[Function]}
        key="6661e4bb-4176-5612-8341-04e2a9379d09"
        tag={
          Object {
            "fields": Object {
              "slug": "/hub-de-conteudo/gluten-free",
            },
            "id": "6661e4bb-4176-5612-8341-04e2a9379d09",
            "name": "gluten_free",
            "tagId": 3,
            "title": "Glúten Free",
          }
        }
        variant={1}
      >
        <li
          className="tags__item tags__item"
        >
          <mockConstructor
            className="tags__link tags__link"
            to="/hub-de-conteudo/gluten-free"
          >
            <a
              className="tags__link tags__link"
              href="/hub-de-conteudo/gluten-free"
            >
              <span>
                Glúten Free
              </span>
            </a>
          </mockConstructor>
        </li>
      </Tag>
      <li
        className="tags__loadMore tags__load-more"
      >
        <Component
          className="tags__button tags__button"
          onClick={[Function]}
        >
          <button
            className="button button tags__button tags__button"
            data-componentname="button"
            data-componentview="button"
            disabled={false}
            hidden={false}
            onClick={[Function]}
            type="button"
          >
            <span>
              Custom button text
            </span>
          </button>
        </Component>
      </li>
    </ul>
  </div>
</Tags>
`;
